Problem with CI on Windows MSVC

After the latest changes in libxml2, the meson-windows-2022.yml CI test fails.

> libLibXml2.a(dict.c.obj) : error LNK2019: unresolved external symbol BCryptGenRandom referenced in function xmlInitRandom

BCryptGenRandom() is called only `#ifdef _WIN32`. The linker does not find it.

@talisein You added the meson-windows-2022.yml CI test. Can you make it work again?
I don't know if the bug is in the CI test file or somewhere else.

--- a/libxml++/meson.build
+++ b/libxml++/meson.build
@@ -103,7 +103,7 @@ if host_machine.system() == 'windows'
   extra_xmlxx_objects += xmlxx_res
 
   if xml2_is_subproject
-    xmlxx_all_deps += [winsock_dep]
+    xmlxx_all_deps += [winsock_dep, bcrypt_dep]
   endif
 endif
 

--- a/meson.build
+++ b/meson.build
@@ -129,6 +129,7 @@ if not xml2_dep.found()
   thread_dep = dependency('threads', required: false)
   zlib_dep = dependency('zlib', required: false)
   winsock_dep = cpp_compiler.find_library('ws2_32', required: false)
+  bcrypt_dep = cpp_compiler.find_library('bcrypt', required: false)
   cmake_build_type = get_option('buildtype')
   if get_option('buildtype') == 'debugoptimized'
     cmake_build_type = 'RelWithDebInfo'

